tests/system_tests_one_router.py

```diff
@@ -242,12 +243,13 @@ def setUpClass(cls):
             ('router', {'mode': 'standalone', 'id': 'QDR'}),
             ('listener', {'port': cls.tester.get_port(), 'role': 'inter-router', 'host': '0.0.0.0'})
         ])
-        cls.router_5 = cls.tester.qdrouterd(name, config_5, wait=False, perform_teardown=False)
+        cls.router_5 = cls.tester.qdrouterd(name, config_5, wait=False, expect=Process.EXIT_FAIL)
 
-        # Give some time for the test to write to the .out file. Without
-        # this sleep, the tests execute too
+        # Give some time for the test to write to the .out file. Without this, the tests execute too
         # fast and find that nothing has yet been written to the .out files.
-        sleep(3)
+        routers: Sequence[Qdrouterd] = (cls.router, cls.router_1, cls.router_2, cls.router_3, cls.router_4, cls.router_5)
+        for router in routers:
+            router.wait(timeout=TIMEOUT)
```

Comment:
This is a good fix. I was wondering if we can catch the [TimeoutExpired](https://docs.python.org/3/library/subprocess.html#subprocess.TimeoutExpired) exception and print the return value of router.wait

Reply:
I was thinking that anything more is not necessary. If the `wait` timeouts, that means the test fails because of it, and the tearDown will print router log that hopefully contains something explanatory. Also, if `router.wait` timeouts, then there is no return value, only exception, no?

```python
>>> import subprocess
>>> subprocess.Popen(["sleep", "2"]).wait(timeout=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/subprocess.py", line 1204, in wait
    return self._wait(timeout=timeout)
  File "/usr/lib64/python3.10/subprocess.py", line 1930, in _wait
    raise TimeoutExpired(self.args, timeout)
subprocess.TimeoutExpired: Command '['sleep', '2']' timed out after 1 seconds
```

Reply:
Ah yes. I wanted to know why the process would not terminate but alas that we cannot figure out when the exception is thrown. Approved.

Reply:
thanks, hopefully that will never happen; and if it does, we cross the bridge then!

Reply:
@ganeshmurthy Actually, if this was `communicate` and not `wait`, in that case it would make sense to print the data carried by the exception. For example, this should be done in https://github.com/skupperproject/skupper-router/issues/226 where it is possible to print the stdout and stderr that the exception carries.

Reply:
> @ganeshmurthy Actually, if this was `communicate` and not `wait`, in that case it would make sense to print the data carried by the exception. For example, this should be done in #226 where it is possible to print the stdout and stderr that the exception carries.

That would be cool. I just saw the following code in https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

```
proc = subprocess.Popen(...)
try:
    outs, errs = proc.communicate(timeout=15)
except TimeoutExpired:
    proc.kill()
    outs, errs = proc.communicate()
```

We could use the above code in system_tests_http2 which passes every test but occasionally fails to terminate and times out at the end. I am not sure why that test fails to terminate. Some process is not terminating is my guess.

Reply:
That's even better than what I had in mind. i had in mind that the `TimeoutExpired` gives access to the stdout/stderr that was fetched so far. It makes better sense to SIGKILL the process first, and then get it all.

In system_tests_http1, there are sometimes threads that are not terminating. For that, I've been thinking that the FakeServer's should be registered with Tester#cleanup, so that they are disposed when the test does tearDown.

Reply:
> 
> In system_tests_http1, there are sometimes threads that are not terminating. For that, I've been thinking that the FakeServer's should be registered with Tester#cleanup, so that they are disposed when the test does tearDown.
Good point. The http2 tests also start a http2 server to test all requests and it should be registered with Tester#cleanup as well. Here is the timeout I am talking about.
https://github.com/skupperproject/skupper-router/runs/5665348873?check_suite_focus=true#step:26:4114

Reply:
communicate in bad_configuration test issue: https://github.com/skupperproject/skupper-router/issues/226
Http1 issue: https://github.com/skupperproject/skupper-router/issues/152
Http2 issue: https://github.com/skupperproject/skupper-router/issues/233